codec/pom.xml

```diff
@@ -100,6 +100,22 @@
       <artifactId>commons-compress</artifactId>
       <scope>test</scope>
     </dependency>
+
+    <dependency>
+      <groupId>org.reflections</groupId>
+      <artifactId>reflections</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.google.code.gson</groupId>
+      <artifactId>gson</artifactId>
+    </dependency>
+    <dependency>
```

Comment:
Ideally we don't want to re-write the metadata generation for every test, but we also don't want to bloat up the non-test sources with test specific code. This creates a test-only dependency to the test-jar of `netty5-transport` where the metadata-generation logic lives